icu4c/source/i18n/decimfmt.cpp

```diff
@@ -797,164 +985,200 @@ EPadPosition DecimalFormat::getPadPosition(void) const {
+        fields->properties->decimalSeparatorAlwaysShown = newValue;
+        touchInternalError();
+    }
 }
 
 UBool DecimalFormat::isDecimalPatternMatchRequired(void) const {
-    return fields->properties->decimalPatternMatchRequired;
+    return U_SUCCESS(internalStatus) && fields->properties->decimalPatternMatchRequired;
 }
 
 void DecimalFormat::setDecimalPatternMatchRequired(UBool newValue) {
-    if (UBOOL_TO_BOOL(newValue) == fields->properties->decimalPatternMatchRequired) { return; }
-    fields->properties->decimalPatternMatchRequired = newValue;
-    touchNoError();
+    if (U_SUCCESS(internalStatus)) {
+        if (UBOOL_TO_BOOL(newValue) == fields->properties->decimalPatternMatchRequired) { return; }
+        fields->properties->decimalPatternMatchRequired = newValue;
+        touchInternalError();
+    }
 }
 
 UBool DecimalFormat::isParseNoExponent() const {
-    return fields->properties->parseNoExponent;
+    return U_SUCCESS(internalStatus) && fields->properties->parseNoExponent;
 }
 
 void DecimalFormat::setParseNoExponent(UBool value) {
-    if (UBOOL_TO_BOOL(value) == fields->properties->parseNoExponent) { return; }
-    fields->properties->parseNoExponent = value;
-    touchNoError();
+    if (U_SUCCESS(internalStatus)) {
+        if (UBOOL_TO_BOOL(value) == fields->properties->parseNoExponent) { return; }
+        fields->properties->parseNoExponent = value;
+        touchInternalError();
+    }
 }
 
 UBool DecimalFormat::isParseCaseSensitive() const {
-    return fields->properties->parseCaseSensitive;
+    return U_SUCCESS(internalStatus) && fields->properties->parseCaseSensitive;
 }
 
 void DecimalFormat::setParseCaseSensitive(UBool value) {
-    if (UBOOL_TO_BOOL(value) == fields->properties->parseCaseSensitive) { return; }
-    fields->properties->parseCaseSensitive = value;
-    touchNoError();
+    if (U_SUCCESS(internalStatus)) {
+        if (UBOOL_TO_BOOL(value) == fields->properties->parseCaseSensitive) { return; }
+        fields->properties->parseCaseSensitive = value;
+        touchInternalError();
+    }
 }
 
 UBool DecimalFormat::isFormatFailIfMoreThanMaxDigits() const {
-    return fields->properties->formatFailIfMoreThanMaxDigits;
+    return U_SUCCESS(internalStatus) && fields->properties->formatFailIfMoreThanMaxDigits;
 }
 
 void DecimalFormat::setFormatFailIfMoreThanMaxDigits(UBool value) {
-    if (UBOOL_TO_BOOL(value) == fields->properties->formatFailIfMoreThanMaxDigits) { return; }
-    fields->properties->formatFailIfMoreThanMaxDigits = value;
-    touchNoError();
+    if (U_SUCCESS(internalStatus)) {
+        if (UBOOL_TO_BOOL(value) == fields->properties->formatFailIfMoreThanMaxDigits) { return; }
+        fields->properties->formatFailIfMoreThanMaxDigits = value;
+        touchInternalError();
+    }
 }
 
 UnicodeString& DecimalFormat::toPattern(UnicodeString& result) const {
-    // Pull some properties from exportedProperties and others from properties
```

Comment:
Here and elsewhere, please use the fail-fast pattern to avoid indentation change.

```cpp
if (U_FAILURE(internalStatus)) {
  return result;
}
// then continue with the old method content
```